test/serializationtests.jl

```diff
@@ -0,0 +1,39 @@
+using Serialization
+@testset "Serialize and Deserialize" begin
+
+    cache = LRU{Int, Int}(maxsize=100_000)
+
+    # Populate the cache with dummy data
+    for i in 1:1_000_000
+        cache[i] = i+1
+    end
+    serialize("cache.jls", cache)
+    deserialized_cache = deserialize("cache.jls")
+    rm("cache.jls")
+
+    # Check that the cache is the same
+    @test cache.maxsize == deserialized_cache.maxsize
+    @test cache.currentsize == deserialized_cache.currentsize
+    @test cache.hits == deserialized_cache.hits
+    @test cache.misses == deserialized_cache.misses
+    @test cache.by == deserialized_cache.by
+    @test cache.finalizer == deserialized_cache.finalizer
+    @test cache.keyset.length == deserialized_cache.keyset.length
+    @test length(cache.dict) == length(deserialized_cache.dict)
```

Comment:
This would test all the same pairs are present in the cache without needing internals:
```suggestion
    @test issetequal(collect(cache), collect(deserialized_cache))
```
(though probably good to check the internal details too, as you do)